scipy/special/functions.json

```diff
@@ -0,0 +1,1359 @@
+{
+    "_cospi": {
+        "_trig.pxd": [
+            "dcospi: d->d",
```

Comment:
Why not make this a key: value pair in JSON:

```
"_trig.pxd": [
    "dcospi": "d->d",
    "ccospi": "D->D"
]
```

Reply:
I guess the order of entries may in principle matter here...

Reply:
It would help if I used valid JSON for my example! Oops.

A working alternative that preserves order would look like:
```
"_trig.pxd": [
    {"dcospi": "d->d"},
    {"ccospi": "D->D"}
]
```

But that's a bit heavyweight for my tastes.

Reply:
I can change it if you like. Longer term it might be nice to change the signature format so that the work of `_parse_signature` is a little easier. (Efforts to make `_generate_pyx` more readable are good IMO.)

Reply:
Let's find out if order matters--I think it will be ok (maybe with some more fixes). This

```
"_trig.pxd": {
    "dcospi": "d->d",
    "ccospi": "D->D"
}
```

would be quite nice.